pure_pursuit_v2:
  ros__parameters:

    # Publishers
    pure_pursuit_lookahead_publisher: "/pure_pursuit/lookahead"
    pure_pursuit_fov_cells_publisher: "/pure_pursuit/fov_cells"
    pure_pursuit_close_wall_cells_publisher: "/pure_pursuit/close_wall_cells"

    # Subscribers 
    pure_pursuit_path_subscriber: "/pure_pursuit/path"
    pure_pursuit_enabled: "/pure_pursuit/enabled"

    # Pure pursuit parameters
    LOOKAHEAD_DISTANCE: 0.18  # m
    WHEEL_BASE: 0.16  # m
    MAX_DRIVE_SPEED: 0.1  # m/s
    MAX_TURN_SPEED: 1.25  # rad/s
    TURN_SPEED_KP: 1.25
    DISTANCE_TOLERANCE: 0.1  # m

    # Obstacle avoidance parameters
    OBSTACLE_AVOIDANCE_GAIN: 0.3
    OBSTACLE_AVOIDANCE_MAX_SLOW_DOWN_DISTANCE: 0.16  # m
    OBSTACLE_AVOIDANCE_MIN_SLOW_DOWN_DISTANCE: 0.12  # m
    OBSTACLE_AVOIDANCE_MIN_SLOW_DOWN_FACTOR: 0.25
    FOV: 200  # degrees
    FOV_DISTANCE: 25  # Number of grid cells
    FOV_DEADZONE: 80  # degrees
    SMALL_FOV: 300  # degrees
    SMALL_FOV_DISTANCE: 10  # Number of grid cells